Framework/DataHandling/inc/MantidDataHandling/ReadMaterial.h

```diff
@@ -0,0 +1,43 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2009 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_DATAHANDLING_ReadMaterial_H_
+#define MANTID_DATAHANDLING_ReadMaterial_H_
+
+#include "MantidKernel/Material.h"
+#include "MantidKernel/MaterialBuilder.h"
+
+#include <map>
+namespace Mantid {
+namespace DataHandling {
+
+/**
+    This class contains code for interpreting a material input for
+   SetSampleMaterial
+*/
+class DLLExport ReadMaterial {
+public:
+  static std::map<std::string, std::string>
+  validateInputs(const std::string &chemicalSymbol, const int z_number,
+                 const int a_number, const double sampleNumberDensity,
+                 const double zParameter, const double unitCellVolume,
+                 const double sampleMassDensity);
```

Comment:
I don't like having so many parameters to a function. could we maybe use a const POD struct maybe called `MaterialParameters` to pass around this this data? This would be particularly useful in the tests where it's hard to tell what each parameter means

Reply:
where would I define the struct for this, in the readMaterial header? or does it need its own since a struct is technically a class and our standard says only one class per header?

Reply:
It's a small helper struct so it can be defined in this header. I would probably make it internal to the `ReadMaterial` class (but not private).